sources/kinesis/__init__.py

```diff
@@ -0,0 +1,70 @@
+import json
+from typing import Optional
+
+import boto3
+import dlt
+from dlt.common import pendulum
+
+from .helpers import KinesisResourceConfiguration
+
+
+@dlt.resource(
+    primary_key=("_shard_id", "_seq_no"),
+    selected=False,
+    spec=KinesisResourceConfiguration,
+)
+def read_kinesis_stream(
+    stream_name: Optional[str] = dlt.secrets.value,
+    aws_access_key_id: Optional[str] = dlt.secrets.value,
+    aws_secret_access_key: Optional[str] = dlt.secrets.value,
+    aws_region: Optional[str] = dlt.secrets.value,
+    last_ts=dlt.sources.incremental(
+        "_kinesis_ts", initial_value=pendulum.now().subtract(seconds=2)
+    ),
+    chunk_size: int = 1000,
+):
+    kinesis_client = boto3.client(
+        "kinesis",
+        aws_access_key_id=aws_access_key_id,
+        aws_secret_access_key=aws_secret_access_key,
+        region_name=aws_region,
+    )
+    shards = kinesis_client.list_shards(StreamName=stream_name)["Shards"]
+    shard_iterators = []
+
+    for shard in shards:
+        shard_id = shard["ShardId"]
+        if timestamp := last_ts.last_value:
+            iterator_params = dict(
+                ShardIteratorType="AT_TIMESTAMP", Timestamp=timestamp
```

Comment:
this is good for testing but not for production. let's build a full version with all the tests and then switch to iterator based on sequence. I had it in my code but it requires to store last value per shard which is possible but tricky with the incremental